### DISCUSSION TITLE
common-tool/arrays

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#99] [Announcements]
# common-tool/arrays

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，Java Arrays工具类的10大常用方法

https://tobebetterjavaer.com/common-tool/arrays.html

**Comment by lilong20065** [comment]
因为数组是一个对象，所以先使用“==”操作符进行判断，如果不相等，再判断是否为 null，两个都为 null，返回 false；紧接着判断 length，不等的话，返回 false；否则的话，依次调用 Objects.equals() 比较相同位置上的元素是否相等。

请问：两个都为 null，返回 false，这里是不是会返回true?

**Reply by qinple** [reply]
我觉得"=="操作的时候null == null 返回true, 不会往后面走了

**Reply by itwanger** [reply]
因为数组是一个对象，所以先使用“==”操作符进行判断，如果不相等，再判断是否为 null，其中一个为 null，返回 false；紧接着判断 length，不等的话，返回 false；否则的话，依次调用 `Objects.equals()` 比较相同位置上的元素是否相等。

参考这个。

**Reply by itwanger** [reply]
我觉得"=="操作的时候null == null 返回true, 不会往后面走了
是这样的

**Comment by akenO8** [comment]
使用 i == iMax -1 有什么问题吗？
三妹明白了什么？
